File: FileUtil.py
import os
import shutil
import zipfile

class FileUtil:

    @staticmethod
    def make(path, content=""):
        if os.path.exists(path):
            return "Error: File already exists."
        with open(path, "w", encoding="utf-8") as f:
            f.write(content)
        return True

    @staticmethod
    def rename(path, new_name):
        if not os.path.exists(path):
            return "Error: Path does not exist."
        directory = os.path.dirname(path)
        new_path = os.path.join(directory, new_name)
        os.rename(path, new_path)
        return new_path

    @staticmethod
    def rename_ext(path, new_ext):
        if not os.path.isfile(path):
            return "Error: File does not exist."
        if not new_ext.startswith("."):
            new_ext = "." + new_ext
        base = os.path.splitext(path)[0]
        new_path = base + new_ext
        os.rename(path, new_path)
        return new_path

    @staticmethod
    def move(src, dst):
        if not os.path.exists(src):
            return "Error: Source does not exist."
        shutil.move(src, dst)
        return True

    @staticmethod
    def compress(folder_path, output_zip):
        if not os.path.isdir(folder_path):
            return "Error: Folder does not exist."
        if not output_zip.endswith(".zip"):
            output_zip += ".zip"
        shutil.make_archive(
            base_name=output_zip.replace(".zip", ""),
            format="zip",
            root_dir=folder_path
        )
        return output_zip

    @staticmethod
    def extract(zip_path, output_dir):
        if not zipfile.is_zipfile(zip_path):
            return "Error: Not a valid ZIP file."
        with zipfile.ZipFile(zip_path, "r") as z:
            z.extractall(output_dir)
        return True

File: FileUtil.java
import java.io.*;
import java.nio.file.*;
import java.util.zip.*;

public final class FileUtil {

    public static boolean make(String path, String content) {
        try {
            File file = new File(path);
            if (file.exists()) return false;
            try (FileWriter writer = new FileWriter(file)) {
                writer.write(content);
            }
            return true;
        } catch (IOException e) {
            return false;
        }
    }

    public static String rename(String path, String newName) {
        File file = new File(path);
        if (!file.exists()) return "Error: Path does not exist.";
        File newFile = new File(file.getParent(), newName);
        if (file.renameTo(newFile)) return newFile.getAbsolutePath();
        return "Error: Rename failed.";
    }

    public static String renameExt(String path, String newExt) {
        File file = new File(path);
        if (!file.isFile()) return "Error: File does not exist.";
        if (!newExt.startsWith(".")) newExt = "." + newExt;
        String base = path.substring(0, path.lastIndexOf('.'));
        File newFile = new File(base + newExt);
        if (file.renameTo(newFile)) return newFile.getAbsolutePath();
        return "Error: Rename failed.";
    }

    public static boolean move(String src, String dst) {
        try {
            Files.move(Paths.get(src), Paths.get(dst), StandardCopyOption.REPLACE_EXISTING);
            return true;
        } catch (IOException e) {
            return false;
        }
    }

    public static boolean compress(String folderPath, String outputZip) {
        try {
            if (!outputZip.endsWith(".zip")) outputZip += ".zip";
            Path zipFile = Paths.get(outputZip);
            try (ZipOutputStream zos = new ZipOutputStream(Files.newOutputStream(zipFile))) {
                Files.walk(Paths.get(folderPath))
                     .filter(Files::isRegularFile)
                     .forEach(p -> {
                         ZipEntry entry = new ZipEntry(Paths.get(folderPath).relativize(p).toString());
                         try {
                             zos.putNextEntry(entry);
                             Files.copy(p, zos);
                             zos.closeEntry();
                         } catch (IOException e) { e.printStackTrace(); }
                     });
            }
            return true;
        } catch (IOException e) {
            return false;
        }
    }

    public static boolean extract(String zipPath, String outputDir) {
        try (ZipInputStream zis = new ZipInputStream(new FileInputStream(zipPath))) {
            ZipEntry entry;
            while ((entry = zis.getNextEntry()) != null) {
                File file = new File(outputDir, entry.getName());
                file.getParentFile().mkdirs();
                try (FileOutputStream fos = new FileOutputStream(file)) {
                    byte[] buffer = new byte[1024];
                    int len;
                    while ((len = zis.read(buffer)) > 0) {
                        fos.write(buffer, 0, len);
                    }
                }
            }
            return true;
        } catch (IOException e) {
            return false;
        }
    }
}
